Drive/rover_drive_system.hpp

```diff
@@ -99,76 +176,165 @@ class RoverDriveSystem
   };
 
   /// Handles the rover movement depending on the mode
-  void HandleModeMovement(
-      units::angle::degree_t roatation_angle,
-      units::angular_velocity::revolutions_per_minute_t speed)
+  /// @param rotation_angle adjusts wheel position depending on mode.
+  /// @param wheel_speed adjusts the movement speed of the rover.
+  void HandleRoverMovement(float roatation_angle, float wheel_speed)
   {
-    if (current_mode_ == Mode::kDrive)
+    units::angle::degree_t angle(roatation_angle);
+    units::angular_velocity::revolutions_per_minute_t speed(wheel_speed);
+    switch (current_mode_)
     {
-      HandleDriveMode();
+      case Mode::kDrive:
+        sjsu::LogInfo("Driving...");
+        HandleDriveMode(speed, angle);
+        break;
+      case Mode::kSpin:
+        sjsu::LogInfo("Spining...");
+        HandleSpinMode(speed);
+        break;
+      case Mode::kTranslation:
+        sjsu::LogInfo("Translating...");
+        HandleTranslationMode(speed, angle);
+        break;
+      default:
+        SetWheelSpeed(kZeroSpeed);
+        sjsu::LogError("Unable to assign drive mode handler!");
+        break;
     }
-    if (current_mode_ == Mode::kSpin)
+    current_speed_ = speed;
+  };
+
+  /// Sets the new driving mode for the rover. Rover will stop before switching
+  /// @param mode Three Modes: D (drive), S (spin), T (translation)
+  void SetMode(char mode = 'S')
+  {
+    switch (mode)
     {
-      HandleSpinMode();
-    }
-    if (current_mode_ == Mode::kTranslation)
+      case 'D':
+      case 'd':
```

Comment:
We should limit this to just one. We shouldn't respect the usage of lower and upper case. The API should be consistent and concise.